src/client/Microsoft.Identity.Client/ManagedIdentityApplication.cs

```diff
@@ -54,5 +54,14 @@ public AcquireTokenForManagedIdentityParameterBuilder AcquireTokenForManagedIden
                 ClientExecutorFactory.CreateManagedIdentityExecutor(this),
                 resource);
         }
+
+        /// <summary>
+        /// Detects and returns the managed identity source available on the environment.
+        /// </summary>
+        /// <returns>Managed identity source detected on the environment if any.</returns>
+        public static ManagedIdentitySource GetManagedIdentitySource()
```

Comment:
how will this behave in a pure FIC scenario?

Reply:
It is considered FIC-agnostic, because it only detects the underlying managed identity environment. It is the calling app to decide whether use a managed identity or FIC.